### ISSUE TITLE
Unresolved execution detection never deletes old tasks if service restarts too often

### ISSUE THREAD

**Issue by GeorgEchterling** [kagkarlsson/db-scheduler#515]
## Expected Behavior

The unresolved execution detection should delete unresolved tasks once they have been *due* for the duration configured in `deleteUnresolvedAfter`. (Default: 14 days)

## Current Behavior

The unresolved execution detection only deletes unresolved tasks after the *local scheduler instance* has detected the unresolved task for the configured duration. If the service never runs for that duration (due to updates or restarts), unresolved tasks are never deleted.

If this is the intended behaviour, it should be documented.

Workaround: I've reduced `deleteUnresolvedAfter` to 3 days, since the service regularly receives updates and never runs for 14 days straight.

## Steps to Reproduce the bug
1. Create a recurring task
2. Delete the task implementation
3. Restart the scheduler
4. Restart the scheduler before the timeout configured in `deleteUnresolvedAfter`
5. The task won't be deleted until `deleteUnresolvedAfter` after the latest restart

## Context
- DB-Scheduler Version : 14.0.2
- Java Version : 21
- Spring Boot (check for Yes) : [x]
- Database and Version : MariaDB 10.11

**Comment by kagkarlsson** [comment]
Yes, you are correct. It should be changed to `age = now - execution-time`

**Comment by kamko** [comment]
Unfortunately, I'm not entirely sure how to implement the check.

Currently, for each unresolved task, we store its `name` and `firstUnresolved` timestamp. If we want to change it to now - execution-time, we need to determine which execution-time to consider. Since there could be multiple instances of a task, using the newest instance should suffice. This is currently done [here](https://github.com/kagkarlsson/db-scheduler/blob/c21b62038c48216176b663461b038f83b4436413/db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/TaskResolver.java#L61-L61), where querying the database for execution times isn't feasible.

Do you have any suggestions on how to approach this? One possibility might be to modify the [dead execution job](https://github.com/kagkarlsson/db-scheduler/blob/c21b62038c48216176b663461b038f83b4436413/db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java#L358-L358). We could then retrieve list of unresolved tasks from the task resolver and then query database for max(execution_time) for calculating the task age.

Would this approach be acceptable?

**Comment by kamko** [comment]
Created PR with basic implementation of proposed idea.

**Comment by kagkarlsson** [comment]
Ah!
